### PR TITLE
Implement jörmungandr txId

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#219 

# Overview

- [x] I have implemented `txId = blake2b256 . putTokenTransfer`
- [x] A golden test
- [x] Auxiliary: I have renamed `putTransaction` to `putSignedTransaction`

# Comments

- I have **not** verified that the golden matches `jcli`

### CODE REVIEW

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs [hunk 1]
```diff
@@ -47,7 +56,22 @@ import qualified Data.ByteString.Char8 as B8
...
+
+addrSpec :: Spec
```
Comment (Anviking) [inline lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs:95]: To preserve horizontal space and keep the `txId` related additions in one place.

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs [hunk 2]
```diff
@@ -47,7 +56,22 @@ import qualified Data.ByteString.Char8 as B8
 import qualified Data.Text as T
 
 spec :: Spec
-spec = describe "EncodeAddress & DecodeAddress" $ do
+spec = do
+    addrSpec
+    describe "txId @(Jormungandr n)" $ do
+        it "(txId largeTx) should match golden" $ do
```
Comment (KtorZ) [inline lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs:72]: Nice!
Would be nice to have some more actually covering various cases:

- Transaction with a single input / output
- Old and new addresses

What do you think ?

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs (hunk 1 above)
Comment (KtorZ) [reply lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs:95]: :+1: , actually for consistency, I'd have also go for `txSpec` :)

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs (hunk 2 above)
Comment (Anviking) [reply lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs:72]: 👍

Comment (Anviking) [reply lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs:72]: Not doing old addresses now as per thread:

https://input-output-rnd.slack.com/archives/GBT05825V/p1560938440087200?thread_ts=1560934314.081600&cid=GBT05825V

**Review by rvl — APPROVED** [review]
Looks good.

I was checking out `putTokenTransfer` (not in this PR) and wondering how we are limiting transaction size before serialisation? Because the maximum number of ins/outs are 255 -- large number, but still a limit.

**Comment by Anviking** [comment]
@rvl thanks, good point 😅  I should have a look at that. I suspect all the uses of `fromIntegral` in the module are potentially very dangerous and that it in this case would overflow.

**Pull request merged**